Lib/shutil.py

```diff
@@ -605,7 +587,7 @@ def _rmtree_unsafe(path, onerror):
         entries = []
     for entry in entries:
         fullname = entry.path
-        if _rmtree_isdir(entry):
+        if entry.is_dir(follow_symlinks=False) and not entry.is_junction():
```

Comment:
```suggestion
        try:
            is_dir = (entry.is_dir(follow_symlinks=False) and
                      not entry.is_junction())
        except OSError:
            is_dir = False
```

The `is_dir()` method call might raise `OSError` on a POSIX platform that lacks the `d_type` field in a directory entry[^1] (e.g. IBM AIX) -- or if the value is `DT_UNKNOWN` because the filesystem doesn't support the `d_type` field. In this case `stat()` has to be called, for which the internal `DirEntry_test_mode()` function only ignores `FileNotFoundError`. We can assume it's not a directory if any `OSError` is raised, just as `os.path.isdir()` does.

[^1]: POSIX doesn't specify `d_type` in a [`dirent` record](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/dirent.h.html). It's found on BSD systems and Linux.